orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -241,9 +243,17 @@ def backward(ctx, *grad_outputs):
                             # input_name is not found in the self._input_names_require_grad list
                             # Append None to results for each input that did not require grad
                             results.append(None)
+
                     # Append gradients of initializer to results
-                    results += [_ort_output_to_torch_tensor(backward_output)
-                                for backward_output in backward_outputs[num_user_input_grads:]]
+                    # Go over each initializer, check if it required grad and append to results accordingly
+                    initializer_index = 0
+                    for inititializer_name in self._onnx_graphs_info.initializer_names:
+                        if inititializer_name in self._onnx_graphs_info.initializer_names_to_train:
```

Comment:
Is this loop O(n^2) in the number of initializers, since `initializer_name in self._onnx_graphs_info.initializer_names_to_train` does linear search in the list? 

Would be good to figure out if we can reduce the complexity and/or cache the mapping work at the same time as export?